### ISSUE TITLE
fix slow page loads

### ISSUE THREAD

**Issue by dertel** [dertel/old-issues#504]
**[Original report](https://bitbucket.org/davidartrecord/artrecord/issues/504) by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

/events/1 for sva gallery, e.g. 

check Logentries for more examples of “request timeout”

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

also maybe check out the memory exceeded errors

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

‌

```diff
diff --git a/app/assets/javascripts/app-include/search.js b/app/assets/javascripts/app-include/search.js
index b95f20f..5bf8782 100644
--- a/app/assets/javascripts/app-include/search.js
+++ b/app/assets/javascripts/app-include/search.js
@@ -1,4 +1,5 @@
 $(document).ready(function(){
+  // account-wide search bar on navigation
 
   $('.search-form').click(function(e){
     e.stopPropagation();
@@ -94,9 +95,13 @@ $(document).ready(function(){
     triggerSelectOnValidInput: false,
     onSearchStart: function(params) {
       $(this).closest('.search').addClass('searching');
+      // $(this).attr('disabled', 'disabled');
+      console.log(this);
     },
     onSearchComplete: function(params) {
       $(this).closest('.search').removeClass('searching');
+      // $(this).removeAttr('disabled');
+      // $(this).focus();
     },
     formatResult: function(suggestion, currentValue){
       return suggestion.formatted
diff --git a/app/controllers/listings_controller.rb b/app/controllers/listings_controller.rb
index c497098..a024e66 100644
--- a/app/controllers/listings_controller.rb
+++ b/app/controllers/listings_controller.rb
@@ -158,13 +158,17 @@ class ListingsController < ApplicationController
     @query = params['query']
 
     if @query.present?
+      helpers.mstimer
 
       @records = PgSearch.
-        multisearch("#{@query} user_id#{current_user.id}").
-        where("pg_search_documents.content ILIKE ?", "user_id#{current_user.id} %").
+        multisearch(@query).
+        where(user_id: current_user.id).
         order(searchable_id: :desc).
-        limit(25).
-        map(&:searchable)
+        limit(25)
+
+      puts @records.count
+      helpers.mstimer
+      @records = @records.map(&:searchable)
 
     else
 
diff --git a/app/models/work.rb b/app/models/work.rb
index 887e9bb..9772109 100644
--- a/app/models/work.rb
+++ b/app/models/work.rb
@@ -3,34 +3,35 @@ class Work < ApplicationRecord
   
   include PgSearch
   multisearchable :against => [
-    :filterable_user_id,
-    :title,
-    :year,
-    :dimensions,
-    :subtype,
-    :value,
-    :pricing_notes,
-    :edition,
-    :description,
-    :status,
-    :notes,
-    :medium,
-    :artist_override,
-    :value,
-    :location,
-    :provenance,
-    :links,
-    :condition,
-    :cost,
-    :inventory,
-    :type,
-    :unique_uid,
-    :uid,
-    :get_filenames_for_search,
-    :owner,
-    :artist_name
-  ], # see /config/intializers/multisearch.rb for more parameters
-  :if => lambda { |record| record.removed_at.nil? }
+      :filterable_user_id,
+      :title,
+      :year,
+      :dimensions,
+      :subtype,
+      :value,
+      :pricing_notes,
+      :edition,
+      :description,
+      :status,
+      :notes,
+      :medium,
+      :artist_override,
+      :value,
+      :location,
+      :provenance,
+      :links,
+      :condition,
+      :cost,
+      :inventory,
+      :type,
+      :unique_uid,
+      :uid,
+      :get_filenames_for_search,
+      :owner,
+      :artist_name
+    ], # see /config/intializers/multisearch.rb for more parameters
+    :if => lambda { |record| record.removed_at.nil? },
+    additional_attributes: -> (rec) { { user_id: rec.user_id } }
 
 
   has_many :work_event_bonds, validate: false
diff --git a/db/schema.rb b/db/schema.rb
index 53c6b98..bd9a7c8 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20190703215949) do
+ActiveRecord::Schema.define(version: 20190910011102) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
@@ -276,6 +276,7 @@ ActiveRecord::Schema.define(version: 20190703215949) do
     t.bigint "searchable_id"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "user_id"
     t.index ["searchable_type", "searchable_id"], name: "index_pg_search_documents_on_searchable_type_and_searchable_id"
   end
```

‌

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

```ruby
class AddUserIdIndexToPgSearch < ActiveRecord::Migration[5.1]
  def change
    add_column :pg_search_documents, :user_id, :integer, index: true
  end
end
```

‌

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

add GIN index to pg_search_documents, see #504

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

fix optimized get_image ordering, see #504

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

speed up Work#formatted_title by using sortable_artist_name, see #504

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

small performance improvements, see #504

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

fix issue with sortable_artist_name, see #504

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

fix issue with sortable_artist_name, see #504
